Codegen/test/TestTypeCobolCodegen.cs

```diff
@@ -1,16 +1,42 @@
 ﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
 using System.Collections.Generic;
 using System.IO;
 using TypeCobol.Codegen.Config;
 using TypeCobol.Codegen.Skeletons;
 using TypeCobol.Compiler; // DocumentFormat
+using TypeCobol.Compiler.Domain;
+using TypeCobol.Compiler.Parser;
+using TypeCobol.Compiler.Symbols;
 using TypeCobol.Tools; // CodeElementDiagnostics
+using TypeCobol.Tools.Options_Config;
+
 
 
 namespace TypeCobol.Codegen {
 
 	[TestClass]
 	public class TestTypeCobolCodegen {
+#if DOMAIN_CHECKER
+        public static TypeCobolConfiguration DefaultConfig = null;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            SymbolTableBuilder.Root = null;
+            //Create a default configurations for options
+            DefaultConfig = new TypeCobolConfiguration();
+            DefaultConfig.Dependencies.Add(Path.Combine(Directory.GetCurrentDirectory(), "resources", "dependencies"));
+            SymbolTableBuilder.Config = DefaultConfig;
+        }
+
+        [TestCleanup]
+        public void TestCleanup()
+        {
+            ProgramSymbolTableBuilder.LastBuilder.RemovePrograms();
```

Comment:
CompilationUnit already call ProgramSymbolTableBuilder.LastBuilder.RemovePrograms();
so why do it again here ?

Reply:
Yes but in CompilationUnit the call is slightly different, the call is PrgSymbolTblBuilder?.RemovePrograms(); and it is for the ProgramSymbolTableBuilder instance of the underlying document, and it removes programs from the previous instance built, thus the first time the previous instance is null.
In case of tests we must cleanup each time a test is ended, because some tests have the same program name, thus symbol will be duplicated from a previous tests otherwise.